sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -175,15 +173,52 @@ class KeyVaultSetting(object):
     """A Key Vault setting.
 
     :ivar str name: The name of the account setting.
-    :ivar str value: The value of the pool setting.
-    :ivar SettingType type: The type specifier of the value.
+    :ivar value: The value of the account setting.
+    :vartype value: str or bool
+    :ivar type: The type specifier of the value.
+    :vartype type: str or KeyVaultSettingType or None
     """
 
-    def __init__(self, **kwargs) -> None:
-        self.name = kwargs.get("name")
-        self.value = kwargs.get("value")
-        self.type = kwargs.get("type")
+    def __init__(
+        self,
+        *,
+        name: str,
+        value: Union[str, bool],
+        setting_type: Optional[Union[str, KeyVaultSettingType]] = None,
+        **kwargs,  # pylint:disable=unused-argument,redefined-builtin
+    ) -> None:
+        self.name = name
+        self.value = value if isinstance(value, str) else str(value)  # `value` is stored as a string
+        self.setting_type = setting_type.lower() if isinstance(setting_type, str) else setting_type
+
+        # If a setting type isn't provided, set it based on `value`'s type (without inferring from the value itself)
+        if self.setting_type is None:
+            if isinstance(value, bool):
+                self.setting_type = KeyVaultSettingType.BOOLEAN
+
+        # If the setting is a boolean, lower-case the string for serialization
+        if self.setting_type == KeyVaultSettingType.BOOLEAN:
+            self.value = self.value.lower()
+
+    def getboolean(self) -> bool:
```

Comment:
Context: I chose this method name to align with [configparser.ConfigParser.getboolean](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.getboolean)